operator/cmd/mesh/install.go

```diff
@@ -110,6 +110,9 @@ func InstallCmd(logOpts *log.Options) *cobra.Command {
 }
 
 func runApplyCmd(cmd *cobra.Command, rootArgs *rootArgs, iArgs *installArgs, logOpts *log.Options) error {
+	if iArgs.revision == "" && cmd.PersistentFlags().Changed("revision") {
+		return fmt.Errorf("empty revision specified")
+	}
```

Comment:
Since we're validating revision here, I think it's a good opportunity to go ahead and validate all of it instead of just empty. Specifically it's a k8s label, so it needs to adhere to the [Syntax and character set](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set). We also state that it can't contain the "." character although I'm having trouble finding official documentation on what the exact syntax is, and 10 minutes examining the call path didn't lead me to anywhere that we actually check this.

So maybe something like

```
if cmd.PersistentFlags().Changed("revision") {
  err := validateRevision(iArgs.revision); if err != nil {
    return err
  }
  return nil
}

func validateRevision(rev string) error{
  if empty: error (is empty)
  if len > 63: error (too long)
  matched, err := regexp.Match(`[[:alnum:]]{1}[a-z0-9A-Z_-]*[[:alnum:]]{1}`, rev)
  if err handle
  if not matched: error (must start/end with alphanum separated by - or _)
}
```

Also as a nit, I'd prefer this to go in PreRunE on the `ic` Command above instead of runApplyCmd. IIRC this is the standard place for flag validation, and in any case doesn't seem to be part of running the apply cmd.